google/src/main/scala/org/broadinstitute/dsde/rawls/google/GoogleUtilities.scala

```diff
@@ -40,6 +41,11 @@ trait GoogleUtilities extends LazyLogging with InstrumentedRetry with GoogleInst
     case t => when500orGoogleError(t)
   }
 
+  def when500orNonExcludedGoogleError(throwable: Throwable, excludedStatusCodes: Set[StatusCode]): Boolean = throwable match {
```

Comment:
Would it read more clearly to inline this in `getGoogleProject` instead of creating a function? The double negative in the name is a bit difficult to parse. This function also has the potential to be confusing if someone passed 500 in the excluded status codes.

Reply:
Ah yeah, this and the other similar predicates are a bit messy. I will put some thought into ways to clean this up.